File: 백준/Silver/10814. 나이순 정렬/나이순 정렬.py
N = int(input())
member = {}

for i in range(N):
    age, name = input().split()
    age = int(age)
    if age in member:
        member[age].append(name)
    else:
        member[age] = [name]

sorted_ages = sorted(member.keys())

for age in sorted_ages:
    for name in member[age]:
        print(age, name)

File: 백준/Silver/10814. 나이순 정렬/나이순 정렬.java
import java.util.*;
public class Main{
    public static void main(String[]args){
        
        Scanner in = new Scanner(System.in);
        StringBuilder sb = new StringBuilder();
        int N = in.nextInt();
        in.nextLine();
        Map <Integer, List<String>> member = new HashMap<>();
        
        for (int i = 0; i < N; i++){
            String [] info = in.nextLine().split(" ");
            int age = Integer.parseInt(info[0]);
            String name = info[1];
            
            if(member.containsKey(age)){
                member.get(age).add(name);
            }
            else{
                List<String> names = new ArrayList<>();
                names.add(name);
                member.put(age, names);
            }
        }
        List<Integer> sortedAges = new ArrayList<>(member.keySet());
        Collections.sort(sortedAges);
        
        for(int age : sortedAges){
            for(String name : member.get(age)){
                sb.append(age + " " + name + "\n");
            }
        }
        
        System.out.print(sb); 
    }
}